### PR TITLE
feat(modal-dialog-primitive): replace reach dialog with reakit dialog

### PR DESCRIPTION
**Pull request by richbachman**
- [x] Modal Dialog Primitive now uses Reakit Dialog
- [x] Updated Modal Dialog Primitive doc page with new api
- [x] Bumped Reakit version in Reakit-Library
- [x] Updated snapshots for Menu and Menu Primitive

### CODE REVIEW

File: packages/paste-core/primitives/modal-dialog/src/index.tsx [hunk 1]
```diff
@@ -1,22 +1,20 @@
...
- */
+/* Keeping the Reach export until we transition Modal to use Reakit */
 import {
   DialogContent as ModalDialogPrimitiveContent,
   DialogOverlay as ModalDialogPrimitiveOverlay,
 } from '@reach/dialog';
 
 export {ModalDialogPrimitiveContent, ModalDialogPrimitiveOverlay};
+
+export type {
```
Comment (TheSisb) [inline packages/paste-core/primitives/modal-dialog/src/index.tsx:9]: Waitaminute... we aren't replacing reach dialog in this PR, we're just adding Reakit dialog. I don't think this is a major change?  I do wonder if we should do the full swap in one go though, as having two dialog libraries is kind of odd maybe?

File: .changeset/tame-emus-hear.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+'@twilio-paste/reakit-library': patch
+'@twilio-paste/core': patch
+---
+
+Bumped to the latest Reakit version.
```
[the remark this answers is not in the document]
Comment (richbachman) [reply .changeset/tame-emus-hear.md:6]: Oh... Maybe. I thought changesets would do that for us.

File: packages/paste-core/primitives/modal-dialog/src/index.tsx (hunk 1 above)
Comment (richbachman) [reply packages/paste-core/primitives/modal-dialog/src/index.tsx:9]: For this PR, I didn't want to rebuild Modal yet, so I'm keeping the Reach export there until we rebuild Modal in a separate PR. As part of that one, I'd remove Reach export and dep.

**Comment by richbachman** [comment]
After further investigation, changing to Reakit will causing too large of a breaking change to the current modal implementation.

**Pull request closed without merging**